polkadot/node/core/approval-voting/src/lib.rs

```diff
@@ -63,6 +63,10 @@ use sc_keystore::LocalKeystore;
 use sp_application_crypto::Pair;
 use sp_consensus::SyncOracle;
 use sp_consensus_slots::Slot;
+use std::time::Instant;
+
+// The maximum block we keep track of assignments gathering times.
+const MAX_BLOCKS_WITH_ASSIGNMENT_TIMESTAMPS: u32 = 100;
```

Comment:
Is this relevant only for when there is finality lag? otherwise we would just prune on finality.

Reply:
Yes, relevant just for finality lag, because we prune on finality.